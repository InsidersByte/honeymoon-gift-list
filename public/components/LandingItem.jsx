/* @flow */

import React, { Component } from 'react';
import smoothscroll from 'smoothscroll';
// FIXME:FLOW need to fix import .styl
import css from './LandingItem.styl';

type PropsType = {
    title: string,
    children?: React$Element<any>,
};

export default class LandingItem extends Component {
    props: PropsType;

    container: any;

    scrollTo = () => {
        const container = this.container;
        smoothscroll(container);
    };

    render() {
        const { title, children } = this.props;

        return (
            <section ref={(c) => { this.container = c; }} className={css.root}>
                <h1 className={css.title}>{title}</h1>

                <div className={css.content}>
                    {children}
                </div>
            </section>
        );
    }
}
