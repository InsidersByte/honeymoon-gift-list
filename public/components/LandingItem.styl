.root
    color rgb(154, 156, 167)
    padding 50px 0
    text-align center

    h1, h2, h3, h4, h5, h6
        color rgb(82, 105, 120)

    a
        color rgb(0, 173, 237)
        text-decoration underline

        &:hover
            color rgb(0, 173, 237)

    &:nth-child(even)
        background rgb(0, 173, 237)
        color white

        h1, h2, h3, h4, h5, h6
            color white

        a
            color white

            &:hover
                color white

    &:before
        content ' '
        display table

    &:after
        clear both
        content ' '
        display table

.content
    font-size .9em
    margin 0 25%
    padding-top 10px

    img
        height auto
        max-width 500px
        width 100%

.title
    font-size 2.5em
    text-transform capitalize
    text-align center

@media (max-width 736px)
    .content
        margin 0 5%
