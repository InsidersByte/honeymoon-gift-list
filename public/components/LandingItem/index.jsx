/* @flow */

import React, { Component } from 'react';
import smoothscroll from 'smoothscroll';
import css from './index.styl';

type PropsType = {
  title: string,
  children?: React$Element<any>,
  postContent?: React$Element<any>,
};

export default class LandingItem extends Component<void, PropsType, void> {
  container: any;

  scrollTo = () => {
    const container = this.container;
    smoothscroll(container);
  };

  render() {
    const { title, children, postContent } = this.props;

    return (
      <section
        ref={c => {
          this.container = c;
        }}
        className={css.root}
      >
        <h1 className={css.title}>{title}</h1>

        <div className={css.content}>
          {children}
        </div>

        {postContent}
      </section>
    );
  }
}
