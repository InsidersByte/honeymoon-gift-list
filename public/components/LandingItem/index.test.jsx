import React from 'react';
import { shallow, mount } from 'enzyme';
import smoothscroll from 'smoothscroll';
import LandingItem from './';

jest.mock('smoothscroll');

describe('LandingItem', () => {
    it('should render correctly', () => {
        const title = 'Hello World';

        const wrapper = shallow(
            <LandingItem title={title}>
                <h1>A Heading</h1>
            </LandingItem>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should call smoothscroll when calling scrollTo', () => {
        const title = 'Hello World';

        const wrapper = mount(
            <LandingItem title={title}>
                <h1>A Heading</h1>
            </LandingItem>,
        );

        expect(wrapper.instance().scrollTo).toBeInstanceOf(Function);
        expect(smoothscroll).not.toHaveBeenCalled();

        wrapper.instance().scrollTo();

        expect(smoothscroll).toHaveBeenCalledTimes(1);
        expect(smoothscroll).toHaveBeenCalledWith(wrapper.instance().container);
    });

    it('should render postContent if it exists', () => {
        const title = 'Hello World';

        const wrapper = shallow(
            <LandingItem title={title} postContent={<h1>I come after the content</h1>}>
                <h1>A Heading</h1>
            </LandingItem>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
