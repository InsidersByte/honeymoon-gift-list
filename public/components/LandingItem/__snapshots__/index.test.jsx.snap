// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingItem should render correctly 1`] = `
<section
  className="root"
>
  <h1
    className="title"
  >
    Hello World
  </h1>
  <div
    className="content"
  >
    <h1>
      A Heading
    </h1>
  </div>
</section>
`;

exports[`LandingItem should render postContent if it exists 1`] = `
<section
  className="root"
>
  <h1
    className="title"
  >
    Hello World
  </h1>
  <div
    className="content"
  >
    <h1>
      A Heading
    </h1>
  </div>
  <h1>
    I come after the content
  </h1>
</section>
`;
